### ISSUE TITLE
Unable to parse POST form on callback

### ISSUE THREAD

**Issue by paullen** [markbates/goth#375]
I've implemented a provider with a callback endpoint according to the example implementation, with the exception of `p.Post("/{provider}/callback", ...)` instead of `p.Get("/{provider}/callback", ...)`. 

Everything works perfectly if I remove the `params.Encode()` check because it always returns a key/value pair which is fair given the `/{provider}/callback` endpoint. But how is the form supposed to be parsed if there always is a URL param?

```
var CompleteUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {

...

params := req.URL.Query()

if params.Encode() == "" && req.Method == "POST" {
        // Doesn't enter this portion because params.Encode() gives the output %3Aprovider=adfs
	req.ParseForm()
	params = req.Form
}

...

}
```
I'm new to this so it might just be a very silly overlook on my part. Thanks for any help in advance.

**Comment by wunderkind2k1** [comment]
Hey - this might help you https://github.com/markbates/goth/pull/356/files#diff-7af721a4e5a3fd8edfcb3f884f51ec3bc62c718a6fd1013f85ecb0c9b256691c You need to update the getState function to parse POST request..

**Comment by bentranter** [comment]
Hey @paullen, this should be addressed as of the latest release [v1.67.0](https://github.com/markbates/goth/releases/tag/v1.67.0). I'm going to close for now, but please re-open if you continue to experience this issue.
